components/motor/gpio/loop.go

```diff
@@ -38,149 +23,80 @@ func (m *EncodedMotor) State(ctx context.Context) ([]float64, error) {
-					"constant_val": 0.0,
-				},
-			},
-			{
-				Name: "trapz",
-				Type: "trapezoidalVelocityProfile",
-				Attribute: rdkutils.AttributeMap{
-					"kpp_gain":   0.45,
-					"max_acc":    30000.0,
-					"max_vel":    4000.0,
-					"pos_window": 0.0,
-				},
-				DependsOn: []string{"set_point", "endpoint"},
-			},
-			{
-				Name: "PID",
-				Type: "PID",
-				Attribute: rdkutils.AttributeMap{
-					"int_sat_lim_lo": -255.0,
-					"int_sat_lim_up": 255.0,
-					"kD":             d,
-					"kI":             i,
-					"kP":             p,
-					"limit_lo":       -255.0,
-					"limit_up":       255.0,
-					"tune_method":    "ziegerNicholsPI",
-					"tune_ssr_value": 2.0,
-					"tune_step_pct":  0.35,
-				},
-				DependsOn: []string{"sum"},
-			},
-			{
-				Name: "gain",
-				Type: "gain",
-				Attribute: rdkutils.AttributeMap{
-					"gain": 0.00392156862,
-				},
-				DependsOn: []string{"PID"},
-			},
-			{
-				Name: "endpoint",
-				Type: "endpoint",
-				Attribute: rdkutils.AttributeMap{
-					"motor_name": m.Name().ShortName(),
-				},
-				DependsOn: []string{"gain"},
-			},
-			{
-				Name: "derivative",
-				Type: "derivative",
-				Attribute: rdkutils.AttributeMap{
-					"derive_type": "backward1st1",
-				},
-				DependsOn: []string{"endpoint"},
-			},
-		},
-		Frequency: 100.0,
+	// convert the motor config ControlParameters to the control.PIDConfig structure for use in setup_control.go
+	convertedControlParams := []control.PIDConfig{{
+		Type: "",
+		P:    m.cfg.ControlParameters.P,
+		I:    m.cfg.ControlParameters.I,
+		D:    m.cfg.ControlParameters.D,
+	}}
+
+	// auto tune motor if all ControlParameters are 0
+	// since there's only one set of PID values for a motor, they will always be at convertedControlParams[0]
+	if convertedControlParams[0].P == 0.0 &&
+		convertedControlParams[0].I == 0.0 &&
+		convertedControlParams[0].D == 0.0 {
+		options.NeedsAutoTuning = true
 	}
 
-	if p == 0.0 && i == 0.0 && d == 0.0 {
-		// when tuning (all PID are zero), the loop has to exclude the trapz block, so the sum block is
-		// temporarily changed to depend on the constant block instead of the trapz block to pass over trapz
-		conf = setSumBlock(conf, "set_point")
-	} else {
-		// if not tuning, set the sum block to once again depend
-		// on the trapz block instead of the constant bloc
-		conf = setSumBlock(conf, "trapz")
+	pl, err := control.SetupPIDControlConfig(convertedControlParams, m.Name().ShortName(), options, m, m.logger)
+	if err != nil {
+		return err
 	}
 
-	return conf
+	m.controlLoopConfig = pl.ControlConf
+	m.loop = pl.ControlLoop
+	m.blockNames = pl.BlockNames
+
+	return nil
 }
 
-func setSumBlock(conf control.Config, firstDependency string) control.Config {
-	sumBlock := control.BlockConfig{
-		Name: "sum",
-		Type: "sum",
-		Attribute: rdkutils.AttributeMap{
-			"sum_string": "+-",
-		},
-		DependsOn: []string{firstDependency, "derivative"},
+func (m *EncodedMotor) startControlLoop() error {
+	// create control loop
```

Comment:
Mild preference to delete this comment. It doesn't add anything that I couldn't get from either line 90 or line 92 already.